### PR TITLE
Add docs wc

### PR DESCRIPTION
**Pull request by Fionoble**
### Background

This PR primarily uses the .dev doc generator + our own custom bash script to scaffold the docs.

### What's in this PR

- `*.doc.ts` files for every component in the Admin surface of UI Extensions
- Adds missing examples to admin components
- Adds `*.doc.ts` for APIs for admin ui extensions
- Adds targets (f.k.a. extension-points) for both block and action extensions
- Adds static page json files for the dev-docs for admin extensions

### Checklist

- [x] I have :tophat:'d these changes
- [x] I have updated relevant documentation

### CODE REVIEW

File: packages/ui-extensions/src/surfaces/admin/extension-points.ts [hunk 1]
```diff
@@ -14,6 +14,46 @@ export interface ExtensionPoints {
...
+    AnyComponent
+  >;
+  'admin.customer-details.block.render': RenderExtension<
+    StandardApi<'admin.customer-details.block.render'>,
+    AnyComponent
+  >;
+
+  // Actions
+  'admin.product-details.action.render': RenderExtension<
+    StandardApi<'Playground'>,
```
Comment (MitchLillie) [inline packages/ui-extensions/src/surfaces/admin/extension-points.ts:34]: Is `Playground` right here? And the same for the rest of the entries.

File: packages/ui-extensions/src/surfaces/admin/extension-points.ts [hunk 2]
```diff
@@ -14,6 +14,46 @@ export interface ExtensionPoints {
...
+    AnyComponent
+  >;
+  'admin.order-index.action.render': RenderExtension<
+    StandardApi<'Playground'>,
+    AnyComponent
+  >;
+  'admin.customer-index.action.render': RenderExtension<
+    StandardApi<'Playground'>,
+    AnyComponent
+  >;
```
Comment (MitchLillie) [inline packages/ui-extensions/src/surfaces/admin/extension-points.ts:56]: I think we're missing the bulk action extension points for action extensions. I'm not sure what their names should be because they're also missing from the spreadsheet

File: packages/ui-extensions/docs/surfaces/admin/staticPages/configuration.doc.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        },
+        {
+          name: 'extensions.targeting',
+          description:
+            'This is a grouping of extension targets that the extension is targeting.',
+          value: 'string',
+          isOptional: false,
+        },
+        {
+          name: 'module',
```
Comment (MitchLillie) [inline packages/ui-extensions/docs/surfaces/admin/staticPages/configuration.doc.ts:71]: I'm not sure if it's feasible, but it would be good to structure this page in a way that makes it clear that `handle` explicitly belongs underneath `extensions.targeting` (and the same for all the other nested properties listed here). For now I think this is fine

**Review by MitchLillie — COMMENTED** [review]
This is a lot, thanks Fio! A couple comments and questions

File: packages/ui-extensions/src/surfaces/admin/extension-points.ts (hunk 1 above)
Comment (Fionoble) [reply packages/ui-extensions/src/surfaces/admin/extension-points.ts:34]: It's not lol, i goofed it. Will update 🙂

File: packages/ui-extensions/src/surfaces/admin/extension-points.ts (hunk 2 above)
Comment (Fionoble) [reply packages/ui-extensions/src/surfaces/admin/extension-points.ts:56]: Good catch!

File: packages/ui-extensions/src/surfaces/admin/extension-points.ts (hunk 1 above)
Comment (Fionoble) [reply packages/ui-extensions/src/surfaces/admin/extension-points.ts]: Updated

File: packages/ui-extensions/docs/surfaces/admin/staticPages/configuration.doc.ts (hunk 3 above)
Comment (Fionoble) [reply packages/ui-extensions/docs/surfaces/admin/staticPages/configuration.doc.ts:71]: Yea, I agree. I think the folks working on the unified toml will have more detailed docs and I'll mimic the pattern they use for it here once that is ready/ships!

Comment (vividviolet) [reply packages/ui-extensions/docs/surfaces/admin/staticPages/configuration.doc.ts:71]: Do we need to repeat the docs again here instead of linking to it https://github.com/Shopify/shopify-dev/pull/34566/files#diff-c23276940c260d6e77eee8a6a207e1d39c801fa802e385d0f3bc9a5b148ef2fe? These are all standard UI extensions config, nothing specific to Admin.

File: packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+    },
+    {
+      type: 'GenericAccordion',
+      title: 'Custom Protocols',
+      sectionContent:
+        "We've added some custom protocols to make it easier to construct URLs to navigate to common localtions.",
+      anchorLink: 'customProtocols',
+      accordionContent: [
+        {
+          title: 'shopify:admin',
```
Comment (tiffanytse) [inline packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts:107]: @johndcollett is it possible to wrap these in a `<code>` element so they don't get capitalized as titles since they're reference to actual code strings? 

Other option would be to think about the titles more like titles; 

`Admin protocol`, `App protocol`, `Extension protocol`, and `Relative URLs`

and in the body use a code element in the description so it's more obvious what the protocol is.

File: packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+            title: 'Get Product Data',
+          },
+        ],
+      },
+    },
+    {
+      type: 'GenericAccordion',
+      title: 'Custom Protocols',
+      sectionContent:
+        "We've added some custom protocols to make it easier to construct URLs to navigate to common localtions.",
```
Comment (tiffanytse) [inline packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts:103]: Update to actionable language and less self referential.

`Custom protocols make it easier to navigate to common locations, and construct URLs.`

File: packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+  body: JSON.stringify(simpleProductQuery),
+});`,
+              },
+            ],
+          },
+        },
+        {
+          title: 'app:',
+          description:
+            'Use the app protocol when you want to construct a URL for your app. Shopify will handle constructing the base URL for your app. This works for both embedded and non-embedded apps.',
```
Comment (tiffanytse) [inline packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts:132]: `Use the app protocol to construct a URL for your app. Shopify will handle constructing the base URL for your app. This works for both embedded and non-embedded apps.`

File: packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+                language: 'tsx',
+                code: '<Link to="app:settings/advanced" />',
+              },
+            ],
+          },
+        },
+        {
+          title: 'extension:',
+          description:
+            'You can use this protocol to trigger an action extension from a block extension.',
```
Comment (tiffanytse) [inline packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts:147]: `Triggers an action extension from a block extension`

- I would add a sentence about why someone might want to use this; or an example use case

File: packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+                language: 'tsx',
+                code: '<Link to={`extension:${extension.handle}`} />',
+              },
+            ],
+          },
+        },
+        {
+          title: '/relative/urls',
+          description:
+            'Relative urls will be relative to the extension. This is useful when you want to link to a route within your extension.',
```
Comment (tiffanytse) [inline packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts:162]: `Relative urls will be relative to the extension you are creating. This is useful when you want to link to a route within your extension`

File: packages/ui-extensions/docs/surfaces/admin/staticPages/configuration.doc.ts [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+  tabletDarkImage: '/assets/landing-pages/templated-apis/hero-dark.png',
+  mobileImage: '/assets/landing-pages/templated-apis/hero.png',
+  mobileDarkImage: '/assets/landing-pages/templated-apis/hero-dark.png',
+  id: 'configuration',
+  sections: [
+    {
+      type: 'Generic',
+      anchorLink: 'how-it-works',
+      title: 'How it works',
+      sectionContent: `You define properties for your admin UI extension in the extension configuration file. The shopify.extension.toml file contains the extension's configuration, which include the extension name, api_version, extension targets and modules.
```
Comment (tiffanytse) [inline packages/ui-extensions/docs/surfaces/admin/staticPages/configuration.doc.ts:19]: Should this be `Admin block extension`?

Comment (tiffanytse) [reply packages/ui-extensions/docs/surfaces/admin/staticPages/configuration.doc.ts:19]: Remove the word `You` and capitalize `Define`

File: packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+          title: 'Developer Preview',
+          sectionContent:
+            'Please not that Admin Action and Block Extensions are currently in a Developer Preview. You may on see them while developing locally on a development store.',
+          type: 'warning',
+        },
+      ],
+      sectionCard: [
+        {
+          name: 'Getting Started',
+          subtitle: 'Get started building your first Admin Extension.',
```
Comment (tiffanytse) [inline packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts:33]: The subtitle copy is meant to be more of a description of the type of content you're linking to 

Screenshot 2023-07-13 at 12 14 25 PM

-----------------------

**See Hydrogen for a great example of how to structure these;** 

https://shopify.dev/docs/api/hydrogen-react


Screenshot 2023-07-13 at 12 17 44 PM
Screenshot 2023-07-13 at 12 17 40 PM

**Review by tiffanytse — COMMENTED** [review]
I haven't gone through all the content but mostly just the home page and left a few suggestions and rewrites; 

Things to avoid: using words like "We". 

Things to do before shipping: 
- Grep for the word shopify; and make sure it's capitalized when in content.
- Look for instances where it makes sense to markup content with inline code elements for easier reading.

File: packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts (hunk 5 above)
Comment (tiffanytse) [reply packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts:103]:
```suggestion
        "Custom protocols make it easier to navigate to common locations, and construct URLs.",
```

File: packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts (hunk 6 above)
Comment (tiffanytse) [reply packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts:132]:
```suggestion
            'Use the app protocol to construct a URL for your app. Shopify will handle constructing the base URL for your app. This works for both embedded and non-embedded apps.',
```

File: packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts (hunk 7 above)
Comment (tiffanytse) [reply packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts:147]:
```suggestion
            'Triggers an action extension from a block extension.',
```

File: packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts (hunk 8 above)
Comment (tiffanytse) [reply packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts:162]:
```suggestion
            'Relative urls will be relative to the extension you are creating. This is useful when you want to link to a route within your extension.',
```

File: packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+            title: 'CLI',
+          },
+        ],
+      },
+    },
+    {
+      type: 'Generic',
+      title: 'Direct API access',
+      sectionContent:
+        "You can now make requests to shopify directly from your extension. You can simply use fetch! The access scopes you have in your extension will be the same as the access scopes you have in your app. If you need to make a request to a resource that requires a different access scope, you will update your app's scopes.",
```
[the remark this answers is not in the document]
Comment (tiffanytse) [reply packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts:86]:
```suggestion
        "Make requests to Shopify directly from your extension, simply use <code>fetch</code>! The access scopes you have in your extension are the same as the access scopes you have in your app. If you need to make a request to a resource that requires a different access scope, update your app's access scopes.",
```
Combo of mine and Liz's feedback

File: packages/ui-extensions-react/src/surfaces/admin/components/Form/examples/basic-form.example.tsx [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+  Button,
+  Form,
+  TextField,
+} from '@shopify/ui-extensions-react/admin';
+
+render('Playground', () => <App />);
+
+function App() {
+  return (
+    <Form onSubmit={() => console.log('onSubmit event')}>
```
[the remark this answers is not in the document]
Comment (Fionoble) [reply packages/ui-extensions-react/src/surfaces/admin/components/Form/examples/basic-form.example.tsx]: Updated

File: packages/ui-extensions/src/surfaces/admin/components/Form/examples/basic-form.example.ts [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+    root.createComponent(Divider)
+  );
+
+  const button = root.createComponent(
+    Button,
+    {accessibilityRole: 'submit'},
+    'Submit',
+  );
+
+  const form = root.createComponent(
```
[the remark this answers is not in the document]
Comment (Fionoble) [reply packages/ui-extensions/src/surfaces/admin/components/Form/examples/basic-form.example.ts:34]: Updated this as well

File: packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts (hunk 10 above)
Comment (Fionoble) [reply packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts]: Still might be a bit wordy. Any thoughts?
Screenshot 2023-07-13 at 6 44 09 PM

File: packages/ui-extensions/src/surfaces/admin/extension-points.ts [hunk 14]
```diff
@@ -14,6 +14,60 @@ export interface ExtensionPoints {
...
+    StandardApi<'admin.order-index.action.render'>,
+    AnyComponent
+  >;
+  'admin.customer-index.action.render': RenderExtension<
+    StandardApi<'admin.customer-index.action.render'>,
+    AnyComponent
+  >;
+
+  // Bulk Actions
+  'admin.product-index.selection.render': RenderExtension<
```
Comment (MitchLillie) [inline packages/ui-extensions/src/surfaces/admin/extension-points.ts:59]: Noticed these don't line up with Elana's PR: https://github.com/Shopify/web/pull/97985

Should be in this format: `'admin.product-index.selection-action.render'`

File: packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts (hunk 10 above)
Comment (Fionoble) [reply packages/ui-extensions/docs/surfaces/admin/staticPages/admin-extensions.doc.ts]: Updated

File: packages/ui-extensions/src/surfaces/admin/extension-points.ts (hunk 14 above)
Comment (Fionoble) [reply packages/ui-extensions/src/surfaces/admin/extension-points.ts:59]: Good catch
I'll update

**Review by lizkhoo-shop — APPROVED** [review]
Looks great!

**Review by vividviolet — APPROVED** [review]
Thanks for doing this!
